storage/txcache/concurrentMap.go

```diff
@@ -0,0 +1,128 @@
+package txcache
+
+import (
+	"sync"
+)
+
+// This implementation is a simplified version of:
+// https://github.com/ElrondNetwork/concurrent-map
+
+// ConcurrentMap is a thread safe map of type string:Anything.
+// To avoid lock bottlenecks this map is divided to several map chunks.
+type ConcurrentMap struct {
+	noChunks uint32
+	chunks   []*concurrentMapChunk
+}
+
+// concurrentMapChunk is a thread safe string to anything map.
+type concurrentMapChunk struct {
+	maxSize uint32
+	items   map[string]interface{}
+	sync.RWMutex
+}
+
+// NewConcurrentMap creates a new concurrent map.
+func NewConcurrentMap(maxSize uint32, noChunks uint32) *ConcurrentMap {
+	m := ConcurrentMap{
+		noChunks: noChunks,
+		chunks:   make([]*concurrentMapChunk, noChunks),
+	}
+
+	chunkSize := maxSize / noChunks
+	if chunkSize == 0 {
+		chunkSize = 1
+	}
+	if maxSize%noChunks != 0 {
```

Comment:
Indeed, `NewConcurrentMap(41, 42)` would have `chunkSize=2`, which is not ok. It won't happen in practice, but we have to fix this nonetheless.

(I took this implementation: https://github.com/ElrondNetwork/concurrent-map/blob/master/concurrent_map.go)

Reply:
I removed the `maxSize` parameter, since it wasn't really necessary (it is for the `concurrent_map` with FIFO behavior). Capacity is given by `EvictionConfig`.

Reply:
just had a discussion in the office and probably we should measure/maintain the capacity in bytes not number of items. this can come in another pull request.